### DISCUSSION TITLE
IKEA Parasoll 3V conversion

### DISCUSSION THREAD

**Discussion by basilfx** [basilfx/TRADFRI-Hacking#69] [Show and tell]
The [IKEA Parasoll](https://github.com/basilfx/TRADFRI-Hacking/tree/master/teardowns/E2013) runs off a singel AAA battery (either 1.2 V rechargeable, or 1.5 V regular).

I wanted to convert the sensor to run off a CR2032 coin cell battery, because I can mount it inside a window frame and hide the sensor. I need the total package to be less than 10 mm thick, which does not work with a AAA battery. All of the components seem to be capable of handling 3 V.

First of all, here is the schematic on paper (I cannot guarantee it is correct). The changes in red are discussed below.

<img src="https://github.com/basilfx/TRADFRI-Hacking/assets/815976/a9bc7a1d-a173-4d8d-85e3-12d9ad88c1ca" height="320" />

I found out that the Parasoll has an on-board DC-DC converter ([OC6811](http://www.sun-ic.com/static/upload/file/20230727/1690447942343157.pdf)), which converts the 1.2 V - 1.5 V into 2.2 V. When the input voltage exceeds 2.5 V, the output voltage (VCC) jumps to approximately 2.4 V. This has some issues with the Hall-effect sensor working improperly. 

The Hall-effect sensor outputs approximately half of VCC at 'rest'. It either goes up or down, depending on the magnetic pole. When VCC rises, the voltage of the Hall-effect sensor rises too (measured at PA00). I believe IKEA uses an ADC or analog comparator with an internal fixed reference voltage. This reference voltage does not rise when VCC rises, which means that the sensor does not detect open/close properly anymore.

Above becomes even more problematic when VCC rises to 3 V. I don't want to use the on-board DC-DC converter, because I believe it is not necessary when running of a higher voltage. The DC-DC converter is ~93% efficient. Without changing the firmware, the only way to mitigate the problem, is to lower the voltage at rest, so it is somewhat equal to original. Luckily, the board has some unpopulated pads, which makes this possible.

Here are the changes I did to make this board run off a 3 V coin cell. I basically added a voltage divider. I do think that the battery level will be misreported, but that is something I do not care too much about.

* R12: replace 0 Ohm resistor with 100 K resistor.
* R13: add 330 K resistor.

Optional: 
* U3: remove, but connect input of L4 (closest to the two capacitors) with output of U3 (closest to the two capacitors)
* R7: replace 200 Ohm with 1 K value to reduce brightness of LED.

About the battery life: I have to see what this will be.

* A single AAA (rechargeable) has 750 mAh at 1.2 V, which is 0.9 Wh. 
* A single CR2032 has 250 mAh at 3 V, which is 0.75 Wh. 
* A single CR2450 has 540 mAh at 3 V, which is 1.6 Wh.

With the DC-DC converter removed, I believe the difference in capacity is not that bad. I have the space to add more coin cells.

**Comment by basilfx** [comment]
Here is a picture of the prototype mounted inside the window frame. The magnet is attached to a metal part on the windows itself.

![3B407407-0B44-4DB6-B8FA-F2CACC76B0B4_1_105_c](https://github.com/basilfx/TRADFRI-Hacking/assets/815976/3a52d176-4079-461b-ac48-7265ce3bb602)

**Comment by MattWestb** [comment]
Have you trying putting one normal diode in serial with the battery and you shall getting 2.3V or 2 it shall being 1.6V ?
If working you dont need doing so much patching of components.
Also you shall getting one OK batter reading with only little high but working detection without changing R3 or putting more resistors for getting it down to PA08.

**Reply by basilfx** [reply]
I have considered this, but I don't have enough electrical engineering knowledge to tell if this is better or not. I believe some power gets lost via the diode, whereas I now limit the current even more that gets to the ADC. 

Also, the hall-effect sensor is only on for a brief moment, five times per second.

**Reply by MattWestb** [reply]
"Normal" diodes have 0.7 V in constant forward voltage drop and the current its going thru is the chip / circuit is using so the power loss is P=V*I for getting the power "burned" of the diode.
(linear power regulating is always "burning" the power and is being lost)

Its common used with pixel LED that is driven by 3.3 V MCU and the LED strips chips is 5 V to putting one diode for the power of the **first** LED chip and its can taking the 3.3 V data signal and the next can doing the normal 5 V OK.

If not patching the PA08s resistor you is very likely getting strange battery readings and also 255 sent from the device (= not valid reading) then its out or range (or burning the chip IO with to high input voltage).

Also i hope you is taking away the DC-DC step up converter so its not making problems and burning the power of you battery.

One PM: Vallhorn motion sensor is is having problems with alkaline batteries not the Zigbee module but the PIR chip is not liking it and doing false detection but i dont knowing if its one MG21 or one MG24 chip as in your device and if its having one DC-DC converter for both chips or only the Zigbee module (its also having one internal DC-DC for the radio and analogue part on the chip if i remember right).

**Reply by basilfx** [reply]
Thanks for the explanation. I think time will tell if my solution holds up. I do believe that all the parts are rated for 3 V. I know the microcontroller is, and I know the design is heavily based on the guidelines by Silicon Labs.

And yes, I took away the DC-DC converter :-)

**Reply by basilfx** [reply]
I have a [Power Profiler Kit II](https://www.nordicsemi.com/Products/Development-hardware/Power-Profiler-Kit-2) and did some energy consumption tests. For these tests, I bought another Parasoll, and did not modify it. For the diode test I used a 1N4148 (I measured a drop of about 0.6 V).

During the tests below, I held the MCU in reset (shorting RESET and GND), so tests are comparable because the current draw of the MCU will not be based on the code it executes.

* Unmodified @ 1.2 V: 61.8 mC -> 0.0206 mWh
* Unmodified @ 1.5 V: 52.4 mC -> 0.0218 mWh
* Unmodified @ 3 V: 26.3 mC -> 0.022 mWh
* Unmodified + diode @ 3 V: 29.9 mC -> 0.0249 mWh
* Modified @ 3V: 29.3 mC -> 0.0244 mWh

Then I repeated the tests, but this time with a paired sensor that I gave some time to get into deep sleep.

* Unmodified @ 1.2 V: 1.52 mC -> 0.000507 mWh
* Unmodified @ 1.5 V: 1.45 mC -> 0.000604 mWh
* Unmodified @ 3 V: 0.70 mC -> 0.000583 mWh
* Unmodified + diode @ 3 V: unreliable, could not get it to sleep (5 mC +)
* Modified @ 3V: 0.62 mC -> 0.000517 mWh

I re-ran several tests, just to make sure, but not all of them. Take the results with a grain of salt. I also think 1 minute is too short to give meaningful numbers, but that's the maximum the tool can aggregate.

This is how the tool looks like:

<img width="1513" alt="Scherm­afbeelding 2024-06-05 om 22 46 09" src="https://github.com/basilfx/TRADFRI-Hacking/assets/815976/1ecfd2c6-c382-4ec6-a396-55201026d10b">

**Reply by MattWestb** [reply]
Interesting and its logical that the current is going down then the voltage is going up then its very likely one switched voltage converter in the chip and if it was linear it shall have being constant and have loosing much more power then at higher voltage.
Its strange that its not going in sleep with diode so shall being interesting if its sending some status change on the network or only "jazzing around" instead of sleeping. 
Of your schematics they have 2 caps before and also after the regulator so shall being good.
But i can thinking of 2 reasons for it:
1: Not clear DC with some ripple that is doing strange things with the hall element sensor (can being very high frequency for bench PSU ?).
2: The hall element sensor is detecting some magnetic feld from cables around or the diode legs (try connecting the diode 90° around or away from the sensor) and the MCU is trying "decoding it".

The power profile kit looks working the same as Silabs WSDK i having but i have not trying the power profiling with it but it shall working well if reading  the Silabs forums,

**Comment by basilfx** [comment]
I finalized the conversion and chose a CR2450. Here is an updated design.

![IMG_9411](https://github.com/basilfx/TRADFRI-Hacking/assets/815976/d3f0b0cc-3485-466f-97da-c900a3bcf5de)
![IMG_9413](https://github.com/basilfx/TRADFRI-Hacking/assets/815976/48433e5b-f5b2-49a0-a3f6-cd7f6f24a43f)

**Reply by PerMejdal** [reply]
It looks good. It hope you will write a summary/guide, if someone else want to replicate your project.

**Reply by MattWestb** [reply]
The only not so good thing is that the PCB antenna is not super but i think its OK.
I hope you not done some metal screw that is interfere / blocking the RF for the antenna (i think you have one hole for screw and the second is for the reset / paring button = very OK).
Also if will getting the all package smaller i think its possible getting some mm by rapping the plastic from the magnet.

One PM: In the gen 3 2 button swishes (On/Off Open/Close Scene) FCC papers they is writing they is using 2 different DC-DC chips but i dont knowing if its the same with this MG24 board.

**Reply by basilfx** [reply]
The antenna is on top, the fastener in the middle (just above the battery). The case is made of PLA plastic (1.5mm thick).

I don't notice any issues with range, even when the window is closed (wooden frame with metal band). Time will tell how stable this is and how much this affects battery life, but so far so good.

**Comment by PerMejdal** [comment]
I find myself replaing the batteries in Parasoll way too often. I am using IKEAs AAA rechargeable 750 mAh batteries.
I was thining about using 2 x AA, and bypass the DC2DC converter. But I could just run them in paralle.

**Reply by MattWestb** [reply]
I think its not one hardware problem then i have seen the same with some of my PARASOLLs.
For first and second gen IKEA controllers it was some bugs in the Silabs SKD that was making the chip not going in to sleep after jumping to one new parent or getting one software lock then writing to the flash but they are fixed now.
But with new MG24 its very likely implanted some new bugs i the the code base but it can also being hardware that is not playing nice.
I have not sniffing the traffic from the device if its pulling its parent all the time and have the transmitter on all the time but defiantly somthing spooky is going on,
2 * AA(A) parallel you is getting 100% more runtime but the best is if we and / or IKEA is knowing way its happening and fixing the problem.

And at the en:
God jul och gott nytt år !!
